raster/src/main/scala/geotrellis/raster/mapalgebra/focal/ZFactorCalculator.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.raster.mapalgebra.focal
+
+import geotrellis.vector.{Extent, Point}
+
+import org.apache.commons.math3.analysis.interpolation._
+
+
+/** Produces a ZFactor for a given point using the prescribed
+ *  conversion method.
+ *
+ *  @param produceZFactor A function that takes a latitude point and
+ *    converts it to its corresponding ZFactor.
+ */
+class ZFactorCalculator(produceZFactor: Double => Double) extends Serializable {
+  /** Produces the associated ZFactor for the given location.
+   *
+   *  @param extent The aread of interest. The center point will be used to determine
+   *    the ZFactor.
+   */
+  def deriveZFactor(extent: Extent): Double =
```

Comment:
Naming, `val zfactor = ZFactorCalculator.deriveZFactor`, the method name can be more terse and meaningful.